### PR TITLE
add 'multiplier' to modify current value

### PR DESCRIPTION
**Pull request by rbo**
Add multiplier to modify current value.

Updated version of #1542 and might be solved #1468 too.

### CODE REVIEW

**Comment by sni** [comment]
Thanks for the patch, this sounds like a good idea.

Took me a while to realize the -f format is only applied when adding thresholds.

```
%> ./check_snmp -H localhost -P 2c -o iso.3.6.1.2.1.1.7.0                                                                                  
SNMP OK - 72 | SNMPv2-MIB::sysServices.0=72 

%> ./check_snmp -H localhost -P 2c -o iso.3.6.1.2.1.1.7.0 -M .1
SNMP OK - 7.200000 | SNMPv2-MIB::sysServices.0=7.200000 

%> ./check_snmp -H localhost -P 2c -o iso.3.6.1.2.1.1.7.0 -M .1 -f "%.2f"
SNMP OK - 7.200000 | SNMPv2-MIB::sysServices.0=7.200000

%> ./check_snmp -H localhost -P 2c -o iso.3.6.1.2.1.1.7.0 -M .1 -f "%.2f" -w 1
SNMP WARNING - *7.20* | SNMPv2-MIB::sysServices.0=7.20;1 
```

File: plugins/tests/check_snmp.t [hunk 1]
```diff
@@ -268,3 +268,9 @@ like($res->output, '/SNMP WARNING - \d+ \*-4\* | iso.3.6.1.4.1.8072.3.2.67.10=\d
 $res = NPTest->testCmd( "./check_snmp -H 127.0.0.1 -C public -p $port_snmp -o .1.3.6.1.4.1.8072.3.2.67.10,.1.3.6.1.4.1.8072.3.2.67.17 -w 1,2 -c 1" );
 is($res->return_code, 2, "Multiple OIDs with some thresholds" );
 like($res->output, '/SNMP CRITICAL - \*\d+\* \*-4\* | iso.3.6.1.4.1.8072.3.2.67.10=\d+c;1;2 iso.3.6.1.4.1.8072.3.2.67.17=-4;;/', "Multiple OIDs with thresholds output" );
+
+$res = NPTest->testCmd( "./check_snmp -H 127.0.0.1 -C public -p $port_snmp -o .1.3.6.1.2.1.25.2.2.0 -M .125 ");
+is($res->return_code, 0, "Multiply OK" );
+
```
[the remark this answers is not in the document]
Comment (rbo) [reply plugins/tests/check_snmp.t:275]: Agree, added more tests.

File: plugins/tests/check_snmp.t [hunk 2]
```diff
@@ -268,3 +268,9 @@ like($res->output, '/SNMP WARNING - \d+ \*-4\* | iso.3.6.1.4.1.8072.3.2.67.10=\d
...
+$res = NPTest->testCmd( "./check_snmp -H 127.0.0.1 -C public -p $port_snmp -o .1.3.6.1.2.1.25.2.2.0 --multiplier=.0009765625 -f '%.3f' ");
+is($res->return_code, 0, "Multiply format OK" );
```
[the remark this answers is not in the document]
Comment (rbo) [reply plugins/tests/check_snmp.t]: Agree, added more tests.

**Comment by rbo** [comment]
@sni rebase on master tree and improve tests. Please review

File: plugins/tests/check_snmp.t [hunk 3]
```diff
@@ -268,3 +268,19 @@ like($res->output, '/SNMP WARNING - \d+ \*-4\* | iso.3.6.1.4.1.8072.3.2.67.10=\d
...
+is($res->output,'SNMP OK - 42 | iso.3.6.1.4.1.8072.3.2.67.19=42 ', "Test plain value of .1.3.6.1.4.1.8072.3.2.67.1" );
+
+$res = NPTest->testCmd( "./check_snmp -H 127.0.0.1 -C public -p $port_snmp -o .1.3.6.1.4.1.8072.3.2.67.19 -M .1");
+is($res->return_code, 0, "Test multiply RC" );
+is($res->output,'SNMP OK - 4.200000 | iso.3.6.1.4.1.8072.3.2.67.19=4.200000 ' , "Test multiply .1 output" );
+
+$res = NPTest->testCmd( "./check_snmp -H 127.0.0.1 -C public -p $port_snmp -o .1.3.6.1.4.1.8072.3.2.67.19 --multiplier=.1 -f '%.2f' ");
+is($res->return_code, 0, "Test multiply RC + format" );
+is($res->output, 'SNMP OK - 4.200000 | iso.3.6.1.4.1.8072.3.2.67.19=4.200000 ', "Test multiply .1 output + format" );
+
```
Comment (sni) [inline plugins/tests/check_snmp.t:283]: Thanks for the test case. But is this the expected behaviour? Setting `%.2f` format string and getting `4.200000`. I would expect `4.20` or und error telling me i used the plugin in a wrong way.

Comment (rbo) [reply plugins/tests/check_snmp.t:283]: I don't know, just use the code from @wnieder. I try to find out why the format is only applied with a threshold but I was to dump. :-(

**Review by sni — APPROVED** [review]

**Comment by sni** [comment]
thanks, should be fine now.

**Comment by rbo** [comment]
@sni thank you for going the last mile!

**Pull request merged**